src/test/java/seedu/address/logic/parser/add/AddCommandParserTest.java

```diff
@@ -77,117 +135,256 @@ public void parse_allAddProjectFieldsPresent_success() {
+                new AddSkillCommand(expectedSkill));
+
+        // multiple item levels - last level accepted
+        assertParseSuccess(parser,
+                ITEM_TYPE_SKILL + PREFIXED_NAME_REACT + PREFIXED_INTERMEDIATE + PREFIXED_BASIC + PREFIXED_TAG_FRONTEND,
+                new AddSkillCommand(expectedSkill));
+
+        expectedSkill = new SkillBuilder(REACT).withTags(VALID_TAG_FRONTEND, VALID_TAG_JAVA).build();
+
+        // multiple item tags - all tags accepted
+        assertParseSuccess(parser,
+                ITEM_TYPE_SKILL + PREFIXED_NAME_REACT + PREFIXED_BASIC + PREFIXED_TAG_FRONTEND + PREFIXED_TAG_JAVA,
+                new AddSkillCommand(expectedSkill));
     }
-     */
 
-    /*
+
     @Test
     public void parse_optionalFieldsMissing_success() {
-        // zero tags
-        Item expectedItem = new PersonalDetailBuilder(AMY).withTags().build();
-        assertParseSuccess(parser, NAME_DESC_AMY + PHONE_DESC_AMY + EMAIL_DESC_AMY + ADDRESS_DESC_AMY,
-                new AddCommand(expectedItem));
+        Internship expectedInternship = new InternshipBuilder(GOOGLE).withTags().build();
+
+        // 0 tags
+        assertParseSuccess(parser,
+                ITEM_TYPE_INTERNSHIP + PREFIXED_NAME_GOOGLE + PREFIXED_ROLE_FRONTEND + PREFIXED_TIME_FROM
+                        + PREFIXED_TIME_TO + PREFIXED_INTERNSHIP_DESCRIPTION,
+                new AddInternshipCommand(expectedInternship));
+
+        Project expectedProject = new ProjectBuilder(ORBITAL).withTags().build();
+
+        // Standard
+        assertParseSuccess(parser, ITEM_TYPE_PROJECT + PREFIXED_NAME_ORBITAL + PREFIXED_TIME_ORBITAL
+                + PREFIXED_WEBSITE_ORBITAL + PREFIXED_DESCRIPTION_ORBITAL,
+                new AddProjectCommand(expectedProject));
+
+        Resume expectedResume = new ResumeBuilder(ME_RESUME).withTags().build();
+
+        // 0 tags
+        assertParseSuccess(parser,
+                ITEM_TYPE_RESUME + PREFIXED_NAME_ME,
+                new AddResumeCommand(expectedResume));
+
+
+        Skill expectedSkill = new SkillBuilder(REACT).withTags().build();
+
+        // 0 tags
+        assertParseSuccess(parser,
+                ITEM_TYPE_SKILL + PREFIXED_NAME_REACT + PREFIXED_BASIC,
+                new AddSkillCommand(expectedSkill));
+    }
+
+    @Test
+    public void parse_compulsoryItemTypeMissing_failure() {
+        // missing item prefix
+        assertParseFailure(parser, PREFIXED_NAME_GOOGLE + PREFIXED_ROLE_FRONTEND + PREFIXED_TIME_FROM
+                        + PREFIXED_TIME_TO + PREFIXED_INTERNSHIP_DESCRIPTION + PREFIXED_TAG_FRONTEND,
+                Item.MESSAGE_CONSTRAINTS);
+
+        // missing item prefix
+        assertParseFailure(parser,
+                INVALID_TYPE_DESC + PREFIXED_NAME_GOOGLE + PREFIXED_ROLE_FRONTEND + PREFIXED_TIME_FROM
+                        + PREFIXED_TIME_TO + PREFIXED_INTERNSHIP_DESCRIPTION + PREFIXED_TAG_FRONTEND,
+                "Not a valid item type!");
```

Comment:
I think this should be an `// invalid item prefix` because `INALID_TYPE_DESC` is `i/ skii`